src/go/rpk/pkg/config/weak.go

```diff
@@ -352,13 +352,19 @@ func (rpc *RedpandaNodeConfig) UnmarshalYAML(n *yaml.Node) error {
 	if err := n.Decode(&internal); err != nil {
 		return err
 	}
+
+	// redpanda won't recognize rpc_server_tls if is a list.
+	v := reflect.ValueOf(internal.Other["rpc_server_tls"])
+	if v.Kind() == reflect.Slice {
+		fmt.Fprintf(os.Stderr, "WARNING: redpanda.yaml's redpanda.rpc_server_tls property is a list, and redpanda reads this as a single element. We cannot automatically fix this because it may break a running cluster. Please check our documentation for how to fix this on live clusters\n")
```

Comment:
Can this be be wrapped in a global sync.Once, so that we ensure we write this to the CLI only _once_?

Also let's change the messaging to...
```
WARNING: Due to an old rpk bug, your redpanda.yaml's redpanda.rpc_server_tls property is a array, while redpanda reads the field as a dictionary. rpk cannot automatically fix this: servers will not be able to rejoin the cluster during a rolling upgrade. To enable TLS on broker RPC ports, you must turn off your cluster, switch the redpanda.rpc_server_tls field to a dictionary, and then turn your cluster back on. This message will continue to appear while redpanda.rpc_server_tls exists and is an array\n
```

We can remove the need to refer to our online documentation.

Reply:
I changed list to an array, which more users might be familiar with, and expanded the wording to avoid any mention of needing to visit online docs.

If you change "list" to "array" here, the test below will need to be updated as well (since it is looking for the word "list")

Reply:
Done :+1: